zavod/zavod/helpers/sanctions.py

```diff
@@ -10,12 +10,17 @@
 
 
 def lookup_sanction_program_key(
-    context: Context, source_key: Optional[str]
+    context: Context,
+    source_key: Optional[str],
+    ignore_source_key: Optional[List[str]] = None,
 ) -> Optional[str]:
-    """Lookup the sanction program key based on the source key."""
+    """Lookup the sanction program key based on the source key, unless it is ignored."""
+    if ignore_source_key and source_key in ignore_source_key:
+        return None
+
     program_key = context.lookup_value("sanction.program", source_key)
     if program_key is None:
-        context.log.warn(f"Program key {program_key!r} not found.")
+        context.log.warn(f"Program key for {source_key!r} not found.")
     return program_key
```

Comment:
another way to do this might be to include the ignored programs in the `sanction.programs` lookup with value `null`.

If we change this to `context.lookup` instead of `context.lookup_value` we can see that the lookup was None for the warning, but treat a value of `None` as _ignore_

Reply:
Yeah that also works.

The value of doing this via the lookup is to be a bit more explicit about data mapping in the metadata file which is a nice property.

Reply:
Ah, that's a good point actually. Yeah, I like your suggestion!